sql/core/src/test/resources/sql-tests/inputs/pgSQL/join.sql

```diff
@@ -0,0 +1,2079 @@
+--   where coalesce(thousand,123) = q1 and q1 = coalesce(hundred,123);
+
+select q1, unique2, thousand, hundred
+  from int8_tbl a left join tenk1 b on q1 = unique2
+  where coalesce(thousand,123) = q1 and q1 = coalesce(hundred,123);
+
+-- explain (costs off)
+-- select f1, unique2, case when unique2 is null then f1 else 0 end
+--   from int4_tbl a left join tenk1 b on f1 = unique2
+--   where (case when unique2 is null then f1 else 0 end) = 0;
+
+select f1, unique2, case when unique2 is null then f1 else 0 end
+  from int4_tbl a left join tenk1 b on f1 = unique2
+  where (case when unique2 is null then f1 else 0 end) = 0;
+
+--
+-- another case with equivalence clauses above outer joins (bug #8591)
+--
+
+-- explain (costs off)
+-- select a.unique1, b.unique1, c.unique1, coalesce(b.twothousand, a.twothousand)
+--   from tenk1 a left join tenk1 b on b.thousand = a.unique1                        left join tenk1 c on c.unique2 = coalesce(b.twothousand, a.twothousand)
+--   where a.unique2 < 10 and coalesce(b.twothousand, a.twothousand) = 44;
+
+select a.unique1, b.unique1, c.unique1, coalesce(b.twothousand, a.twothousand)
+  from tenk1 a left join tenk1 b on b.thousand = a.unique1                        left join tenk1 c on c.unique2 = coalesce(b.twothousand, a.twothousand)
+  where a.unique2 < 10 and coalesce(b.twothousand, a.twothousand) = 44;
+
+--
+-- check handling of join aliases when flattening multiple levels of subquery
+--
+
+-- explain (verbose, costs off)
+-- select foo1.join_key as foo1_id, foo3.join_key AS foo3_id, bug_field from
+--   (values (0),(1)) foo1(join_key)
+-- left join
+--   (select join_key, bug_field from
+--     (select ss1.join_key, ss1.bug_field from
+--       (select f1 as join_key, 666 as bug_field from int4_tbl i1) ss1
+--     ) foo2
+--    left join
+--     (select unique2 as join_key from tenk1 i2) ss2
+--    using (join_key)
+--   ) foo3
+-- using (join_key);
+
+
+-- [SPARK-28377] Fully support correlation names in the FROM clause
+-- select foo1.join_key as foo1_id, foo3.join_key AS foo3_id, bug_field from
+--   (values (0),(1)) foo1(join_key)
+-- left join
+--   (select join_key, bug_field from
+--     (select ss1.join_key, ss1.bug_field from
+--       (select f1 as join_key, 666 as bug_field from int4_tbl i1) ss1
+--     ) foo2
+--    left join
+--     (select unique2 as join_key from tenk1 i2) ss2
+--    using (join_key)
+--   ) foo3
+-- using (join_key);
+
+-- [SPARK-20856] Support statement using nested joins
+--
+-- test successful handling of nested outer joins with degenerate join quals
+--
+
+-- explain (verbose, costs off)
+-- select t1.* from
+--   text_tbl t1
+--   left join (select *, '***'::text as d1 from int8_tbl i8b1) b1
+--     left join int8_tbl i8
+--       left join (select *, null::int as d2 from int8_tbl i8b2) b2
+--       on (i8.q1 = b2.q1)
+--     on (b2.d2 = b1.q2)
+--   on (t1.f1 = b1.d1)
+--   left join int4_tbl i4
+--   on (i8.q2 = i4.f1);
+
+-- select t1.* from
```

Comment:
skip this case for correlation names, too?

Reply:
No. for nested joins.